app/models/cashbox.rb

```diff
@@ -0,0 +1,7 @@
+class Cashbox < ActiveRecord::Base
+  has_many :deposits, class_name: 'Payment', foreign_key: 'destination_id'
+  has_many :withdrawls, class_name: 'Payment', foreign_key: 'source_id'
```

Comment:
What's the advantage of having separate `deposits` & `withdrawals` associations? Why not just have `:payments` (actually, I would have preferred `:transactions` - perhaps we can still make this change?) whose value is either negative or positive? It doesn't seem right that to get all transactions/payments for a single cashbox, we need to query for both deposits & withdrawals separately. In this case, if there is still some utility in being able to retrieve just the deposits or just the withdrawals, we can create scopes that filter transactions based on whether their value is positive or negative.